src/parser.rs

```diff
@@ -3065,11 +3065,26 @@ impl<'a> Parser<'a> {
     }
 
     /// Parse a LIMIT clause
-    pub fn parse_limit(&mut self) -> Result<Option<Expr>, ParserError> {
+    pub fn parse_limit(&mut self) -> Result<Option<Limit>, ParserError> {
         if self.parse_keyword(Keyword::ALL) {
             Ok(None)
         } else {
-            Ok(Some(Expr::Value(self.parse_number_value()?)))
+            let first_number = self.parse_number_value()?;
+            let second_number = if  self.consume_token(&Token::Comma){
```

Comment:
Since this is specific to the mysql dialect, can you please add a guard here like

```
if dialect_of!(self is MySqlDialect) {
}
```